modules/docs/docgen/spec/enhancedComponentParser.spec.ts

```diff
@@ -0,0 +1,720 @@
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.properties.0.type.value', 'string');
+      expect(symbols).toHaveProperty('0.type.parameters.1.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.parameters.1.name', 'useHook1');
+      expect(symbols).toHaveProperty('0.type.parameters.2.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.parameters.2.name', 'useHook2');
+    });
+
+    it('should handle "composeHooks" with empty elemProps', () => {
+      const program = createProgramFromSource(`
+        export const useMyComponent = composeHooks(
+          createElemPropsHook(useMyModel)(
+            () => ({ id: '' })
+          )
+          useHook1,
+          useHook2
+        )
+
+        // make the parser consider these exported symbols
+        export const useHook1 = undefined
+        export const useHook2 = undefined
+      `);
+
+      const symbols = parse(program, 'test.ts', [enhancedComponentParser]);
+
+      expect(symbols).toHaveProperty('0.name', 'useMyComponent');
+      expect(symbols).toHaveProperty('0.type.kind', 'callExpression');
+      expect(symbols).toHaveProperty('0.type.name.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.name.name', 'composeHooks');
+      expect(symbols).toHaveProperty('0.type.parameters.0.kind', 'function');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.0.kind', 'parameter');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.0.name', 'model');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.0.type.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.0.type.name', 'MyModel');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.1.kind', 'parameter');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.1.name', 'elemProps');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.1.type.kind', 'object');
+      expect(symbols).toHaveProperty('0.type.parameters.0.parameters.1.type.properties.length', 0);
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.kind', 'object');
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.properties.0.kind', 'property');
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.properties.0.name', 'id');
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.properties.0.type.kind', 'primitive');
+      expect(symbols).toHaveProperty('0.type.parameters.0.returnType.properties.0.type.value', 'string');
+      expect(symbols).toHaveProperty('0.type.parameters.1.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.parameters.1.name', 'useHook1');
+      expect(symbols).toHaveProperty('0.type.parameters.2.kind', 'symbol');
+      expect(symbols).toHaveProperty('0.type.parameters.2.name', 'useHook2');
+    });
+  });
+
+  it('should handle "composeHooks" a subModel hook', () => {
```

Comment:
Reads weird, I think this should be:
```suggestion
  it('should handle "composeHooks" with a subModel hook', () => {
```